### PR TITLE
Key shortcut for enabling/disabling mouse buttons (F10)

### PR DESCRIPTION
**Pull request by Stedoss**
Adds a F10 shortcut to toggle mouse buttons, I tried to make it act and look like it does on the live version. Also displays a message showing the current state of the mouse buttons (whether they're enabled or not).

### CODE REVIEW

File: osu.Game/Overlays/OnScreenDisplay.cs [hunk 1]
```diff
@@ -114,11 +115,21 @@ public OnScreenDisplay()
...
-        private void load(FrameworkConfigManager frameworkConfig)
+        private void load(FrameworkConfigManager frameworkConfig, OsuConfigManager osuConfig)
         {
             trackSetting(frameworkConfig.GetBindable<FrameSync>(FrameworkSetting.FrameSync), v => display(v, "Frame Limiter", v.GetDescription(), "Ctrl+F7"));
             trackSetting(frameworkConfig.GetBindable<string>(FrameworkSetting.AudioDevice), v => display(v, "Audio Device", string.IsNullOrEmpty(v) ? "Default" : v, v));
             trackSetting(frameworkConfig.GetBindable<bool>(FrameworkSetting.ShowLogOverlay), v => display(v, "Debug Logs", v ? "visible" : "hidden", "Ctrl+F10"));
+            trackSetting(osuConfig.GetBindable<bool>(OsuSetting.MouseDisableButtons), v => 
+            {
+                string currentMouseState = "";
+                if (osuConfig.Get<bool>(OsuSetting.MouseDisableButtons))
```
Comment (Jorolf) [inline osu.Game/Overlays/OnScreenDisplay.cs]: did you look at the other "examples" ?

**Comment by TPGPL** [comment]
Dont use your master branch for contributing

**Comment by Jorolf** [comment]
also compare it to this:
image

File: osu.Game/Overlays/OnScreenDisplay.cs [hunk 2]
```diff
@@ -114,11 +115,12 @@ public OnScreenDisplay()
         }
 
         [BackgroundDependencyLoader]
-        private void load(FrameworkConfigManager frameworkConfig)
+        private void load(FrameworkConfigManager frameworkConfig, OsuConfigManager osuConfig)
         {
             trackSetting(frameworkConfig.GetBindable<FrameSync>(FrameworkSetting.FrameSync), v => display(v, "Frame Limiter", v.GetDescription(), "Ctrl+F7"));
             trackSetting(frameworkConfig.GetBindable<string>(FrameworkSetting.AudioDevice), v => display(v, "Audio Device", string.IsNullOrEmpty(v) ? "Default" : v, v));
             trackSetting(frameworkConfig.GetBindable<bool>(FrameworkSetting.ShowLogOverlay), v => display(v, "Debug Logs", v ? "visible" : "hidden", "Ctrl+F10"));
+            trackSetting(osuConfig.GetBindable<bool>(OsuSetting.MouseDisableButtons), v => display(!v, "Mouse Buttons", v ? "Disabled" : "Enabled", "F10"));
```
Comment (smoogipooo) [inline osu.Game/Overlays/OnScreenDisplay.cs]: Can we rename the config setting to "GameplayMouseButtons" with default value true? Which would also clean up the whole `v => !v, v` stuff going on in here

**Comment by peppy** [comment]
Gameplay options shouldn't be displayed on the global OSD; there will be a local one for these

**Comment by peppy** [comment]
We now already have the ability to disable mouse buttons by *unbinding* them.

I think rather than adding these options, the correct approach is to make "input profiles" which can be customised.

**Pull request closed without merging**